### ISSUE TITLE
Amazon S3 Two-way Replication via Replica Modification Sync

### ISSUE THREAD

**Issue by ewbankkit** [hashicorp/terraform-provider-aws#16539]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

> Amazon S3 Replication now gives you the flexibility of replicating object metadata changes for two-way replication between buckets.
> Amazon S3 replica modification sync can help you keep object metadata such as tags, ACLs, and Object Lock settings replicated between replicas and source objects.

### New or Affected Resource(s)

* [`aws_s3_bucket`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket)

### Potential Terraform Configuration

```hcl
resource "aws_s3_bucket" "source" {
  bucket = "example"

  versioning {
    enabled = true
  }

  replication_configuration {
    role = aws_iam_role.replication.arn

    rules {
      id     = "foobar"
      prefix = "foo"
      status = "Enabled"

      source_selection_criteria {
        replica_modifications {
          Status = "Enabled"
        }
      }

      destination {
        bucket        = aws_s3_bucket.destination.arn
        storage_class = "STANDARD"
      }
    }
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/12/amazon-s3-replication-adds-support-two-way-replication/).
[Developer Guide](https://docs.aws.amazon.com/AmazonS3/latest/dev/replication-for-metadata-changes.html).

**Comment by Kanikamiglani31** [comment]
Hello team, Do we have this feature working in terraform? I am looking for both "Delete marker replication" & "Replica modification sync" in terraform.

**Comment by Kanikamiglani31** [comment]
Hey all,

Meanwhile this can be achieved from `local-exec` and `template_file` with aws-cli.

The solution is present in the answer below:

https://stackoverflow.com/questions/67729643/how-can-we-enable-amazon-s3-replication-modification-sync-in-terraform/67792274#67792274

**Comment by tmccombs** [comment]
> Meanwhile this can be achieved from local-exec and template_file with aws-cli.

As I've mentioned on other issues, that workaround doesn't work very well if you use a provider that assumes an IAM role.

**Comment by anGie44** [comment]
Noting here that while not available in the S3 bucket resource, `replication_modifications` is available in the standalone resource `aws_s3_bucket_replication_configuration` , available since `v3.66.0` of the Terraform AWS Provider. Latest argument reference available here: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_replication_configuration#source_selection_criteria
